lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,126 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { PackageType, datasource, defaultRegistryUrl } from './common';
+
+export class HexpmBobDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? defaultRegistryUrl}:${packageName}`,
+  })
```

Comment:
What's a reasonable time to cache these results for? The default is 30 minutes.

Reply:
@JamieMagee Usually there’s one release per approximately one week available.

It takes them 1-2 hours between the release and the completed build.

I think anything under 1-2 hours would be acceptable.